### ISSUE TITLE
[META] Dedicated "Security" section in the admin UI

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#3624]
I would like us to have a dedicated "Security" admin page and a respective Admin menu item in order to:

- show that we take security seriously
- allow site admins to perform security-related tasks in one single place

Ideas for this page include:

- settings for blocking users (failed login attempts, bots etc.)
- provide a default security.txt file ((https://www.drupal.org/project/securitytxt)), and an easy way to edit that info via the UI. The default, out of the box text should point to contact details for reporting security issues in b.org, but site owners should be able to change that if required by their organization (same as we allow them to edit the maintenance mode message)
- settings for protecting forms from spam (expose honeypot settings etc.)
- link to our https://backdropcms.org/security page

Other ideas:
- A "Security overview" block for the dashboard (#495)
- ???

**Comment by jenlampton** [comment]
Interesting. I've also been thinking that I wanted to move "Development" up to the top level...  how would you feel about "Development" > "Security"  (or admin/development/security) for the security page?

**Comment by klonos** [comment]
> I've also been thinking that I wanted to move "Development" up to the top level

Yeah, this is already done if you install the devel module, resulting in 2 "Development" menu items, a top-level one, and one under "Configuration". I think that we should merge these, and I agree that we should move "Development" to become top level.

> how would you feel about "Development" > "Security"

Hmm, I would actually prefer "Security" under "Configuration", because that does not seem to be an exclusively dev-related topic to me.

In fact, I would also rather that "Performance" was moved out of "Development", and then have both "Security" and "Performance" live under "Configuration" > "System"

So, to sum up:

- Development (becomes top-level)
- Configuration
  - System
    - Security (new menu item - this issue here)
    - Performance (moved out of "Development" - separate issue)
    - Updates (#3388)

**Comment by jenlampton** [comment]
> and then have both "Security" and "Performance" live under "Configuration" > "System"

What's the value of `System` here?

**Comment by jenlampton** [comment]
> What's the value of System here? 

I just want to add here what I mentioned in [the issue about the updates link](https://github.com/backdrop/backdrop-issues/issues/3388) ...

I object to using system because that's sort of a cop-out "other" category, where we put things that we can't figure out where else to put. I never know what's in there, and I never look there for anything important.

If we do start to put things there that aren't "Other" things, we should really define what `System` means first so we know if we are using it correctly :)

**Comment by klonos** [comment]
Fair enough, how about this then?:

- Development (becomes top-level: #3652)
- Configuration
  - ...
  - Security (new menu item - this issue here)
  - System
  - Updates (#3388)
  - ...

**Comment by jenlampton** [comment]
Better!

**Comment by klonos** [comment]
I have added https://www.drupal.org/project/security_review to the list of ideas:

![image](https://user-images.githubusercontent.com/2423362/60621556-53e2e980-9e21-11e9-985d-b4c55c5078fd.png)

**Comment by klonos** [comment]
Here's what's in the recent v5.4 of WordPress:

![screencapture-demos2-softaculous-WordPresskmfw797kdo-wp-admin-site-health-php-2020-07-16-05_00_49](https://user-images.githubusercontent.com/2423362/87618040-89ebc680-c721-11ea-967b-86b0e81d3ddd.png)

Passing checks are collapsed by default, to reduce clutter on that page. Here's a screenshot of what other checks are included, with some of them expanded:

![Screen Shot 2020-07-16 at 5 04 28 am](https://user-images.githubusercontent.com/2423362/87618183-e64ee600-c721-11ea-9c9e-b926b9782c47.png)

**Comment by jenlampton** [comment]
Interesting, I wonder if we could add the same kind of categorization to items on our status report. I suspect we may have way too many categories :) 

I've been using Security review on a handful of my sites, and I'm actually finding it not that helpful. There are a few checks that throw false positives, whereas similar checks we already have in core are (mostly around file permissions) are more accurate. I do think there are a few good ones though, and we should consider incorporating those into our status report.

**Comment by yorkshire-pudding** [comment]
I agree with this proposal. Security review can be  quite helpful, though I think there is [one test that is particularly unhelpful](https://github.com/backdrop-contrib/security_review/issues/2).

Other modules prevent username and/or password enumeration which might be worth mentioning.

**Comment by stpaultim** [comment]
@alanmels just opened a similar issue and closed it, I think it because of this existing issue. 

This seems like a good idea and I think Alan's suggestion is consistent with what was being discussed here. I think we just need a PR to get this issue moving again. 

> One thing seems a bit tricky to me: the security / anti-spam modules in the example are contrib. How can we evaluate in core, which modules belong to that section? For example: do all anti-spam modules belong there? So we might also consider this a contrib candidate?

Isn't the point here to create a section where security / anti-spam modules can list themselves? I think that core can provide the section, but each modules decides on it's own where to put it's menu link. Don't we just want to provide a good place for them to go. Once a section exists, we can submit PR's to obvious candidates for this section to move their menu link. 

I would just like to clarify if this issue is about a "Security Page" or a "Security Section" under Configuration - as the title suggests? These are two different ideas.

**Comment by alanmels** [comment]
I believe this issue should be revived and a new Security section must be implemented.

**Comment by indigoxela** [comment]
So maybe the first things to figure out here:

1. What should go to that section (core settings, modules...)
2. How to get contrib/custom modules in there (hook or create dedicated "tags" for info files?)
3. What it should look like (informative, but not overwhelming...)
4. Maybe clearly define the target audience, so we can properly address them

^^ That's random order and probably incomplete. :wink:

**Comment by alanmels** [comment]
> Maybe the path was chosen by intention. For instance under admin/config/content for content authoring related modules or admin/config/people for user account related modules? Stuffing them all under the same path (also admin menu) could be very confusing.

Right, but where modules like Protected Forms or Honeypot should go if they are not exactly about content authoring? They are about security, spam-prevention.

**Comment by alanmels** [comment]
Not every module implements hook_menu, but those who do could list their configuration links under the Security section. And here are some of the modules which could be listed in that section:

https://backdropcms.org/project/captcha
https://backdropcms.org/project/antibot
https://backdropcms.org/project/riddler
https://backdropcms.org/project/ip_anon
https://backdropcms.org/project/honeypot
https://backdropcms.org/project/rename_admin_paths
https://backdropcms.org/project/spambot
https://backdropcms.org/project/previous_login
https://backdropcms.org/project/hsts
https://backdropcms.org/project/security_review
https://backdropcms.org/project/securitytxt

**Comment by alanmels** [comment]
And I wonder why there is separate Search section, when _Search settings_ link could be placed in the _System_ section? Are there also lot's of Web services modules to have their own section? More than the modules about security, protection and spam-fighting? I don't think so. You see, we inherited the structure of the Configuration page (admin/config) from Drupal and never updated it according to new trends, needs. Imo, security and protection modules are more important to have their own section than Web Services with a single _RSS publishing_ link there.

**Comment by BWPanda** [comment]
I'd love to see a new 'Privacy and Security' *section* under Configuration:

![image](https://user-images.githubusercontent.com/2385329/177038243-aa9106bd-d16d-4e4b-837a-fa431857a82d.png)

I think the blocker is not having anything to put there yet... If/when one of the linked features in the OP gets merged, we can add this section and move its UI there. But until then, there's not much point having a blank section of the menu.

**Comment by klonos** [comment]
> I think the blocker is not having anything to put there yet ...

There are some things from the OP that we can put there + I'm hoping that #4696 will get merged soon (it is a Backdrop adaptation of a D7 backport of a feature that was introduced in D9.2.x).

**Comment by klonos** [comment]
Saw this and licked the visual/aesthetics:
![image](https://user-images.githubusercontent.com/2423362/222884208-937f2c38-5dae-4385-b10f-415f218b29fb.png)

Putting it here for inspiration.

**Comment by BWPanda** [comment]
> and licked the visual/aesthetics

How'd they taste? :smile:

**Comment by klonos** [comment]
🤣 🤣 🤣 

Surprised you didn't mention the "fro me the" ...I have my days with typing, don't I?

**Comment by yorkshire-pudding** [comment]
> Oh how issue threads grow. I haven't read everything yet. But, what are the pros and cons of honeypot over antibot module? Should this be a separate issue or a forum question?

@izmeez - forum really; that's not a question about core and not the focus of this issue
